From ce52ced3360fdd37bdefbc8d2004f810cafd65ae Mon Sep 17 00:00:00 2001
From: Mike Yoder <myoder@cloudera.com>
Date: Wed, 18 Feb 2015 12:03:32 -0800
Subject: [PATCH 400/596] CLOUDERA-BUILD CDH-25021 Add the logredactor as a
 dependency, but optional so that that downstream
 projects don't get it. This way packaging will get
 it, but it won't show up in everyone's classpath.

---
 hadoop-common-project/hadoop-common/pom.xml |   10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/hadoop-common-project/hadoop-common/pom.xml b/hadoop-common-project/hadoop-common/pom.xml
index 2023ca6..b9269cb 100644
--- a/hadoop-common-project/hadoop-common/pom.xml
+++ b/hadoop-common-project/hadoop-common/pom.xml
@@ -302,6 +302,16 @@
       <artifactId>bcprov-jdk16</artifactId>
       <scope>test</scope>
     </dependency>
+    <!-- Needed only at runtime when the redacting log4j appender
+         is configured.  Should show up in the common lib/ directory
+         without showing up as a dependency for downstream projects.
+    -->
+    <dependency>
+      <groupId>org.cloudera.logredactor</groupId>
+      <artifactId>logredactor</artifactId>
+      <scope>runtime</scope>
+      <optional>true</optional>
+    </dependency>
   </dependencies>
 
   <build>
-- 
1.7.9.5

